### PR TITLE
Upgrading the netcoreapp2.x and netcoreapp3.0 build configurations to netcoreapp3.1

### PR DESCRIPTION
**Pull request by ViktorHofer**
Upgrading the netcoreapp2.x and netcoreapp3.0 build configurations to netcoreapp3.1
and cleaning up conditions so that future tfm upgrades require less project file changes.

Adding NETStandardError markers to prevent available netstandard2.0 assets being picked
instead, same as already done for System.Speech and others.

For more details please see the reasoning in the linked issue.

Contributes to #53282

### CODE REVIEW

File: src/libraries/System.Data.Odbc/src/System.Data.Odbc.csproj [hunk 1]
```diff
@@ -10,13 +10,14 @@
   <PropertyGroup>
     <IsPartialFacadeAssembly Condition="$(TargetFramework.StartsWith('net4'))">true</IsPartialFacadeAssembly>
     <!-- https://github.com/dotnet/arcade/issues/5717 -->
-    <NoWarn Condition="($([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', '$(NetCoreAppCurrent)'))) and '$(TargetsAnyOS)' == 'true'">$(NoWarn);SA1121</NoWarn>
-    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetFramework)' == 'netstandard2.0' or '$(TargetsAnyOS)' == 'true'">SR.Odbc_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
-  </PropertyGroup>
-  <ItemGroup Condition="$(TargetFramework.StartsWith('$(NetCoreAppCurrent)')) or $(TargetFramework.StartsWith('netcoreapp2.0'))" >
+    <NoWarn Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">$(NoWarn);SA1121</NoWarn>
```
Comment (Anipik) [inline src/libraries/System.Data.Odbc/src/System.Data.Odbc.csproj:13]: i dont think we parse out the rid from tfm for netcoreappCurrent, we will need to keep the check to startsWith

Comment (ViktorHofer) [reply src/libraries/System.Data.Odbc/src/System.Data.Odbc.csproj:13]: Look at the previous condition, this NoWarn should only apply to the rid-less NetCoreAppCurrent.

File: src/libraries/System.IO.Pipelines/ref/System.IO.Pipelines.csproj [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
   <PropertyGroup>
-    <TargetFrameworks>$(NetCoreAppCurrent);netstandard2.0;net461</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent);netcoreapp3.1;netstandard2.0;net461</TargetFrameworks>
     <Nullable>enable</Nullable>
-    <ExcludeFromPackage Condition="'$(TargetFramework)' == 'net461'">true</ExcludeFromPackage>
     <!-- We only plan to use this ref in netcoreapp. For all other netstandard compatible frameworks
     we should use the lib asset instead. -->
-    <PackageTargetFramework>netcoreapp2.0</PackageTargetFramework>
-    <ExcludeCurrentNetCoreAppFromPackage>true</ExcludeCurrentNetCoreAppFromPackage>
+    <ExcludeFromPackage Condition="'$(TargetFramework)' != 'netcoreapp3.1'">true</ExcludeFromPackage>
```
Comment (Anipik) [inline src/libraries/System.IO.Pipelines/ref/System.IO.Pipelines.csproj:7]: why we are building netcoreapp3.1 if we are not shipping it ?  why are adding net6.0 to the package ?

Comment (ViktorHofer) [reply src/libraries/System.IO.Pipelines/ref/System.IO.Pipelines.csproj:7]: We are excluding every ref except the netcoreapp3.1 one.

**Review by Anipik — APPROVED** [review]
couple of comments, other than that looks good

File: src/libraries/System.IO.Pipelines/ref/System.IO.Pipelines.csproj (hunk 2 above)
Comment (Anipik) [reply src/libraries/System.IO.Pipelines/ref/System.IO.Pipelines.csproj:7]: and why are we adding the net6.0 config to the package ?

**Comment by ViktorHofer** [comment]
Failures are:
- https://github.com/dotnet/runtime/issues/53420
- https://github.com/dotnet/runtime/issues/53421
- https://github.com/dotnet/core-eng/issues/13026

**Pull request merged**